### ISSUE TITLE
Large (>1MB) colour profiles can crash with segmentation fault

### ISSUE THREAD

**Issue by goldo** [lovell/sharp#937]
Hi,

We are experiencing node/express server crash when using sharp with specific images (4): `segmentation fault`

I've tried to go from using Buffer to Stream without any change.
I've tried to isolate the problem without doing anything on the image like this: 
```js
const sharp = require('sharp');

sharp('./images/bug.jpg')
  .toFile('./images/out.jpg', (err, info) => {
    console.log({ err, info });
  });
```
here is one of the images:  https://framapic.org/pRRvN4QD9sK4/akysnsC4rj15.jpg

Working locally with OSX Sierra, node 6.11.1, & same problem remotely with google gcp instances.

Thanks for helping

**Comment by lovell** [comment]
Bonsoir, I've had a quick look at the example image and everything seems OK. The following works locally for me on Ubuntu Linux with the latest sharp and Node.js 8:
```javascript
sharp('akysnsC4rj15.jpg').toFile('out.jpg', console.log);
```
I see peak memory usage (RSS) of ~200MB.

Are you able to get any more information about the crash? If you're able to consistently trigger it then a backtrace via a tool such as gdb would be ideal.

**Comment by goldo** [comment]
hi @lovell thanks for your reply. I've tried node 8 without success. Then I saw this issue https://github.com/lovell/sharp/issues/939 talking about installing vips with flags, it worked on my mac ! I need to do the same on Google GCP (using Cent OS) and it should be fine :)

**Comment by lovell** [comment]
@goldo Is this problem still occurring? If so, are you able to get a backtrace?

**Comment by goldo** [comment]
As I said, the problem was libvips. I installed it, reinstalled the node_modules, and it worked. Without installing libvips mannualy, it didn't work with these files..

**Comment by lovell** [comment]
Thanks for the update, I'll close in favour of #939.

**Comment by goldo** [comment]
But it was not specific to osx. Our servers on centos had the exact same problem, they needed vips for some pictures (<0.1% of our images)

**Comment by lovell** [comment]
If you have the time to generate a backtrace of this crash when using sharp without a globally-installed version of libvips then that would be very useful to help debug this further, thank you.

**Comment by goldo** [comment]
with `node-stack-trace` for example ?

**Comment by lovell** [comment]
An OS-level backtrace is required here rather than a JS stacktrace.

As I mentioned above, `gdb` is probably the best tool for this.

You use it by running `gdb --args /usr/bin/node your-app.js`, enter `run` at the prompt, wait for the crash then enter `bt` at the prompt for the backtrace.

**Comment by goldo** [comment]
hi @lovell & sorry for the delay
I couldn't make it work with gdb, so I used lldb and I got this:

```
vips warning: icc_transform: embedded profile incompatible with image
Process 87520 stopped
* thread #7, name = 'worker', stop reason = EXC_BAD_ACCESS (code=1, address=0x3d204d69203a)
    frame #0: 0x0000000102e693aa libglib-2.0.0.dylib`g_mutex_lock + 13
libglib-2.0.0.dylib`g_mutex_lock:
->  0x102e693aa <+13>: movq   (%rbx), %rdi
    0x102e693ad <+16>: testq  %rdi, %rdi
    0x102e693b0 <+19>: je     0x102e693c0               ; <+35>
    0x102e693b2 <+21>: callq  0x102e6caf4               ; symbol stub for: pthread_mutex_lock
Target 0: (node) stopped.
(lldb) bt
* thread #7, name = 'worker', stop reason = EXC_BAD_ACCESS (code=1, address=0x3d204d69203a)
  * frame #0: 0x0000000102e693aa libglib-2.0.0.dylib`g_mutex_lock + 13
    frame #1: 0x00000001050d9791 libvips.42.dylib`vips_area_copy + 18
    frame #2: 0x0000000102ef5ae3 libgobject-2.0.0.dylib`g_boxed_copy + 212
    frame #3: 0x0000000102ef5dbf libgobject-2.0.0.dylib`g_value_set_boxed + 108
    frame #4: 0x00000001050d9fe9 libvips.42.dylib`transform_save_string_blob + 97
    frame #5: 0x0000000102f176b9 libgobject-2.0.0.dylib`g_value_transform + 117
    frame #6: 0x00000001050e57f0 libvips.42.dylib`vips_image_open_input + 1286
    frame #7: 0x00000001050e44d9 libvips.42.dylib`vips_image_build + 550
    frame #8: 0x00000001050dc5dc libvips.42.dylib`vips_object_build + 28
    frame #9: 0x00000001050e38c3 libvips.42.dylib`vips_image_rewind_output + 111
    frame #10: 0x00000001050e2eb0 libvips.42.dylib`vips_image_pio_input + 159
    frame #11: 0x00000001050e65b4 libvips.42.dylib`vips_image_generate + 444
    frame #12: 0x00000001050e2dc5 libvips.42.dylib`vips_image_write + 90
    frame #13: 0x00000001050c0f6b libvips.42.dylib`read_jpeg_image + 418
    frame #14: 0x00000001050c046c libvips.42.dylib`vips__jpeg_read + 170
    frame #15: 0x00000001050c0249 libvips.42.dylib`vips__jpeg_read_file + 145
    frame #16: 0x00000001050c218b libvips.42.dylib`vips_foreign_load_jpeg_file_load + 62
    frame #17: 0x00000001050b41c8 libvips.42.dylib`vips_foreign_load_start + 171
    frame #18: 0x00000001050ed79d libvips.42.dylib`vips__region_start + 116
    frame #19: 0x00000001050eee9b libvips.42.dylib`vips_region_generate + 22
    frame #20: 0x00000001050ee076 libvips.42.dylib`vips_region_fill + 57
    frame #21: 0x00000001050eee38 libvips.42.dylib`vips_region_prepare + 156
    frame #22: 0x00000001050e2f0f libvips.42.dylib`vips_image_write_gen + 31
    frame #23: 0x00000001050eeecf libvips.42.dylib`vips_region_generate + 74
    frame #24: 0x00000001050ee076 libvips.42.dylib`vips_region_fill + 57
    frame #25: 0x00000001050eee38 libvips.42.dylib`vips_region_prepare + 156
    frame #26: 0x00000001050e2f0f libvips.42.dylib`vips_image_write_gen + 31
    frame #27: 0x00000001050eeecf libvips.42.dylib`vips_region_generate + 74
    frame #28: 0x00000001050ef1fc libvips.42.dylib`vips_region_prepare_to_generate + 111
    frame #29: 0x00000001050ef108 libvips.42.dylib`vips_region_prepare_to + 497
    frame #30: 0x00000001050e88e8 libvips.42.dylib`wbuffer_work_fn + 41
    frame #31: 0x00000001050f04cc libvips.42.dylib`vips_thread_main_loop + 261
    frame #32: 0x00000001050efc89 libvips.42.dylib`vips_thread_run + 52
    frame #33: 0x0000000102e4f67b libglib-2.0.0.dylib`g_thread_proxy + 90
    frame #34: 0x00007fffe660e93b libsystem_pthread.dylib`_pthread_body + 180
    frame #35: 0x00007fffe660e887 libsystem_pthread.dylib`_pthread_start + 286
    frame #36: 0x00007fffe660e08d libsystem_pthread.dylib`thread_start + 13
```

hope it helps

**Comment by lovell** [comment]
Brilliant, thank you, it looks like you've run into https://github.com/jcupitt/libvips/commit/7b64246af2cdf9c7b9fae0f29e5483bb6a7376d2 which was fixed in libvips 8.5.6.

This explains why the pre-compiled libvips binaries (8.5.5) fail and brew-installed libvips (8.5.9) works.

**Comment by lovell** [comment]
sharp v0.19.0 with libvips v8.6.1 as a dependency is now available.
